vnext/Microsoft.ReactNative/Modules/PaperUIManagerModule.cpp

```diff
@@ -0,0 +1,793 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include <DynamicReader.h>
+#include <DynamicWriter.h>
+#include <IReactContext.h>
+#include <IReactRootView.h>
+#include <Modules/PaperUIManagerModule.h>
+#include <Modules\NativeUIManager.h>
+#include <Views/ViewManager.h>
+#include <XamlUtils.h>
+#include "ShadowNodeBase.h"
+#include "Unicode.h"
+#include "XamlUIService.h"
+
+namespace Microsoft::ReactNative {
+
+struct ViewAtIndex {
+  int64_t tag;
+  int64_t index;
+  static bool ViewAtIndexCompare(
+      const std::shared_ptr<ViewAtIndex> &x,
+      const std::shared_ptr<ViewAtIndex> &y) noexcept {
+    return (x->index < y->index);
+  }
+};
+
+std::vector<int64_t> PopulateIntVecFromDynamicArray(React::JSValueArray &arr) {
+  std::vector<int64_t> vec(arr.empty() ? 0 : arr.size());
+
+  if (arr.empty())
+    return vec;
+
+  auto i = 0;
+  for (auto &val : arr) {
+    vec[i] = val.AsInt64();
+    ++i;
+  }
+  return vec;
+}
+
+winrt::Microsoft::ReactNative::ReactPropertyId<
+    winrt::Microsoft::ReactNative::ReactNonAbiValue<std::unique_ptr<UIManagerSettings>>>
+UIManagerSettingsProperty() noexcept {
+  static winrt::Microsoft::ReactNative::ReactPropertyId<
+      winrt::Microsoft::ReactNative::ReactNonAbiValue<std::unique_ptr<UIManagerSettings>>>
+      prop{L"ReactNative.UIManager", L"Settings"};
+  return prop;
+}
+
+UIManagerSettings::UIManagerSettings(
+    const std::shared_ptr<facebook::react::MessageQueueThread> batchingUIMessageQueue,
+    std::vector<std::unique_ptr<IViewManager>> &&viewManagers)
+    : batchingUIMessageQueue(std::move(batchingUIMessageQueue)), viewManagers(std::move(viewManagers)) {}
+
+void UIManager::SetSettings(
+    winrt::Microsoft::ReactNative::IReactPropertyBag const &properties,
+    std::unique_ptr<UIManagerSettings> &&settings) noexcept {
+  properties.Set(
+      UIManagerSettingsProperty().Handle(),
+      winrt::Microsoft::ReactNative::ReactNonAbiValue<std::unique_ptr<UIManagerSettings>>{std::in_place,
+                                                                                          std::move(settings)});
+}
+
+winrt::Microsoft::ReactNative::ReactPropertyId<
+    winrt::Microsoft::ReactNative::ReactNonAbiValue<std::weak_ptr<NativeUIManager>>>
```

Comment:
Nit: I've seen a bunch of these.. Macro might be too much but should we add type declaration for this?

soemthing like:
```cpp
template <class T>
using NonAbiPropertyId = winrt::Microsoft::ReactNative::ReactPropertyId<winrt::Microsoft::ReactNative::ReactNonAbiValue<std::weak_ptr<T>>>;
```

to make the decl a little shorter and readible like the ones for regular propertyid's:

```cpp
NonAbiPropertyId <NativeUIManager> NativeUIManagerProperty() noexcept {
  static NonAbiPropertyId<NativeUIManager>  prop{L"ReactNative.UIManager", L"NativeUIManager"};
  return prop;
}
```

to match

```cpp
static const React::ReactPropertyId<bool> &SystemIsRTLPropertyId() noexcept {
  static const React::ReactPropertyId<bool> prop{L"ReactNative.I18n", L"SystemIsRTL"};
  return prop;
}